### PR TITLE
Ticket 16 - add queries to get all articles

### PR DESCRIPTION
**Pull request by Brynster99**

### CODE REVIEW

File: controllers/controllers.js [hunk 1]
```diff
@@ -32,8 +32,16 @@ exports.getTopics = (req, res, next) => {
...
 exports.getArticles = (req, res, next) => {
-  fetchArticles()
-    .then((articles) => {
+  const promises = [
+    fetchArticles(req.query.sort_by, req.query.order, req.query.topic),
+  ];
+
+  if (req.query.topic) {
+    promises.push(checkExists('topics', 'slug', req.query.topic));
+  }
```
Comment (dav3rid) [inline controllers/controllers.js:41]: Big fan of this! Only doing as necessary!!!

File: errors.js [hunk 2]
```diff
@@ -9,6 +9,10 @@ exports.customErrors = (err, req, res, next) => {
 
 exports.psqlErrors = (err, req, res, next) => {
   if (err.code === '22P02') {
+    // 22P02: Invalid input type/syntax
+    res.status(400).send({ msg: 'Bad Request' });
+  } else if (err.code === '42703') {
+    // 42703: Column does not exist
     res.status(400).send({ msg: 'Bad Request' });
```
Comment (dav3rid) [inline errors.js:14]: As your resulting behaviour is identical at this conditional statement, consider grouping together

File: models/models.js [hunk 3]
```diff
@@ -16,12 +16,41 @@ exports.fetchTopics = () => {
...
+    ],
+    order: ['ASC', 'DESC'],
+  };
+
+  if (
+    !allowedValues.sortBy.includes(sortBy) ||
+    !allowedValues.order.includes(order)
+  ) {
+    return Promise.reject({ status: 400, msg: 'Bad Request' });
+  }
```
Comment (dav3rid) [inline models/models.js:39]: Great use of `||` here - keeping things DRY!
Suggested something similar for your error handler

**Review by dav3rid — APPROVED** [review]

File: errors.js (hunk 2 above)
Comment (Brynster99) [reply errors.js:14]: Good point!

**Pull request merged**